src/mergecal/calendar_merger.py

```diff
@@ -33,12 +39,29 @@ def __init__(
 
         self.calendars: list[Calendar] = calendars
 
+        self.template: TYPE_TEMPLATE = (
+            safe_load(template) or {} if isinstance(template, str) else template
+        )
+
     def add_calendar(self, calendar: Calendar) -> None:
         """Add a calendar to be merged."""
         self.calendars.append(calendar)
 
     def merge(self) -> Calendar:
         """Merge the calendars."""
+        # template the calendar
+        env = NativeEnvironment()
+        calendar_template = self.template.get("VCALENDAR", {})
+        calendar_attributes_template = calendar_template.get(  # type: ignore
+            "attributes", {}
+        )
+        for attribute in sorted(calendar_attributes_template.keys()):  # type: ignore
+            attribute_template = calendar_attributes_template[attribute]
+            value = env.from_string(attribute_template).render(
+                calendars=self.calendars, attribute=attribute
+            )
+            self.merged_calendar.add(attribute, value)
+        # add events
         for cal in self.calendars:
             for component in cal.walk("VEVENT"):
```

Comment:
Correct me if wrong but the VEVENT templating would go here?
so as we loop through vevent component we would need to apply the VEVENT template changes?

Reply:
correct! the other components could also be added.